src/param.rs

```diff
@@ -289,10 +289,13 @@ impl AudioParamProcessor {
                         let end_index_clipped = end_index.min(count);
                         let n_values = end_index_clipped - start_index;
 
+                        let mut val = self.value;
```

Comment:
I have cherry picked this commit, made a small change and added a regression test in #19